index.js

```diff
@@ -1,54 +1,39 @@
 const exec = require('child_process').exec
 const createDebugger = require('debug')
 const path = require('path')
+const _ = require('underscore')
 
 const depsCommand = `npm ls --parseable`
 
-function splitOnNewLine(string) {
-  let dbgr = createDebugger(arguments.callee.name)
-  //dbgr(string)
+function mapOutputToDepList(string) {
   return string.split('\n')
 }
 
 function pathOfPackage(string) {
-  // pass dependency chain with
-  let dbgr = createDebugger(arguments.callee.name)
-  //dbgr(string)
   return path.join(string, "package.json")
 }
 
 function packageFor(string) {
-  let dbgr = createDebugger(arguments.callee.name)
-  //dbgr(string)
   return require(string)
 }
 
-function keyFromPackage(key) {
+function keysFromPackage(keys) {
   return object => {
-    let dbgr = createDebugger(arguments.callee.name)
-    //dbgr(JSON.stringify(object))
-
-    return { packageName: object.name, version: object.version, keyValue: object[key] }
+    return _.pick(object, keys)
   }
 }
 
 function tailOfPath(string) {
-  let dbgr = createDebugger(arguments.callee.name)
-  //dbgr(string)
   return string.split('/').pop()
 }
 
 function removeEmptyStrings(item) {
   return item !== ''
 }
 
-function start(key, cb) {
-  let dbgr = createDebugger(arguments.callee.name)
-  dbgr("In start function")
-
+function start(keys, cb) {
   exec(depsCommand, (error, stdout, sterr) => {
-    let depKeys = splitOnNewLine(stdout).slice(1).filter(removeEmptyStrings).map(pathOfPackage).map(packageFor).map(keyFromPackage(key))
-    //dbgr(depKeys)
+    let depKeys = _.chain(mapOutputToDepList(stdout)).tail().filter(removeEmptyStrings).map(pathOfPackage).map(packageFor).map(keysFromPackage(keys)).value()
```

Comment:
Went with tail() using the chain method. Stuck with Underscore since it is already there. Lodash would be better if I was using more functionality. I am still not 100% happy with this but it is slightly better than slice(1).
I tried a couple of different ways to wrap it in a revealing function but it fails to be called as a function.